File: code/Java/第15章 动态规划/15.1 动态规划/StairsRecursive.java
public class StairsRecursive {
    public static void main(String[] args) {
        System.out.println(climbStairs(10));
    }

    public static int climbStairs(int n) {
        if(n <= 0) {
            return 0;
        } else if(n == 1) {
            return 1;
        } else if(n == 2) {
            return 2;
        }
        return climbStairs(n-1) + climbStairs(n-2);
    }
}

File: code/Python/Chapter12/12-1/stairs_recursive.py
def climb_stairs(n):
    if n <= 0:
        return 0
    elif n == 1:
        return 1
    elif n == 2:
        return 2
    return climb_stairs(n-1) + climb_stairs(n-2)

def main():
    print(climb_stairs(10))

if __name__ == '__main__':
    main()